doc/specs/#674 - PowerShell cmdlets.md

```diff
@@ -0,0 +1,443 @@
+
+"For [#674](https://github.com/microsoft/winget-cli/issues/674)"
+
+## Abstract
+
+This specification is to begin drafting PowerShell cmdlets for the Windows Package Manager. @doctordns submitted the issue
+and this specification is intended to capture the intent for the syntax. The PowerShell Team has recently released
+[PowerShell Crescendo Preview.3](https://devblogs.microsoft.com/powershell/announcing-powershell-crescendo-preview-3/). We
+are considering using this framework to help model the syntax and create a module so users can experiment with the syntax.
+Several of the winget.exe commands provide output that is not representative of properly formatted objects, so there will
+likely be gaps to address natively in the COM interface for the Windows Package Manager, or we will provide an argument to output the results from the client to JSON.
+
+## Inspiration
+
+@ThomasNieto was the first to create an Issue for [Native PowerShell Support](https://github.com/microsoft/winget-cli/issues/221).
+This is one of the [most desired features](https://github.com/microsoft/winget-cli/issues?page=1&q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc+label%3AIssue-Feature) for the Windows Package Manager.
+
+## Solution Design
+
+The initial solution will start with [PowerShell Crescendo Preview.3](https://devblogs.microsoft.com/powershell/announcing-powershell-crescendo-preview-3/).
+Many of the major considerations are related to proper PowerShell nouns and verbs. @MilkFather suggested [command aliases](https://github.com/microsoft/winget-cli/issues/380) for the winget.exe
+client and @Jaifroid also suggested [find](https://github.com/microsoft/winget-cli/issues/1299) rather than search.
+
+## UI/UX Design
+
+The main UI/UX differences will be in the command names, parameters, and how the output is formatted for users. For the sake of
+consistency the sub headings below will map to existing commands in winget.exe to begin drafting some syntax proposals. In order
+to keep the discussion focussed, I'll create GitHub discussions for each of these and link them here as we work through driving
+consensus.
+
+The PowerShell module name will be "WindowsPackageManager"
+The Adjective "WinGet" may change, but is used consistently in the cmdlets until such time as we agree on the proper term.
+
+>Note: Naming things is hard, and one of the conventions of PowerShell is to make nouns singular.
+---
+
+### Client command `winget install`
+
+**PowerShell Cmdlet**
+
+```PowerShell
+Install-WinGetPackage
+```
+**Parameters**
+
+-Manifest
+
+-PackageIdentifier
+[Alias("Id")]
+
+-Name
+
+-Moniker
+
+-Version
+
+-Source
+
+-Scope
+
+-Exact
+
+-Interactive
+
+-Silent
+>Note: we may switch the default to "silent" from "silent with progress" in winget.exe so there may be changes here.
+
+-Locale
+
+-Log
+
+-Override
+
+-Location
+
+---
+
+### Client command `winget show`
+
+**PowerShell Cmdlet**
+```PowerShell
+Get-WinGetPackageVersion
+```
+
+-Detail
```

Comment:
it would be good to elaborate what these different parameters do. since the name of the cmdlet includes *version* i'm a little puzzled why you need a *version* parameter